packages/preferences-api/src/browser/preference-protocol.ts

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (C) 2017 Ericsson and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { Event, Disposable } from '@theia/core/lib/common';
+
+export interface PreferenceChangedEvent {
+    changes: PreferenceChange[]
+}
+
+export interface PreferenceChange {
+    readonly preferenceName: string;
+    readonly newValue?: any;
+    readonly oldValue?: any;
+}
+
+export const PreferenceService = Symbol('PreferenceService');
```

Comment:
we should not have the protocol file anymore:
- preference service related concepts should be moved to preference-service.ts
- preference provider related concepts should be moved to preference-provider.ts

Reply:
Also maybe PreferenceProvider is a bit confusing, I mean technically packages provide preferences via configuration defined in `preference-proxy.ts`, so I'm still not sure about the current naming...